### PR TITLE
EPMRPP-36738 Rewrite auto-analysis settings for new api

### PR DESCRIPTION
**Pull request by AmsterGet**

### CODE REVIEW

File: app/src/pages/inside/settingsPage/settingsTabs/autoAnalysisTab/analysisForm/analysisForm.jsx [hunk 1]
```diff
@@ -150,22 +159,24 @@ export class AnalysisForm extends Component {
     this.setAnalysisMode(formInputsValues);
   };
 
-  onFormSubmit = (data) => {
-    const dataToSend = {
+  onFormSubmit = (formData) => {
+    const preparedData = normalizeAttributesWithPrefix(formData, 'analyzer');
```
Comment (DiscoElevator) [inline app/src/pages/inside/settingsPage/settingsTabs/autoAnalysisTab/analysisForm/analysisForm.jsx]: ANALYZER_ATTRIBUTE_PREFIX

File: app/src/pages/inside/settingsPage/settingsTabs/autoAnalysisTab/analysisForm/strategyBlock/strategyBlock.jsx [hunk 2]
```diff
@@ -72,7 +74,7 @@ export class StrategyBlock extends Component {
           fieldWrapperClassName={cx('switcher-wrapper')}
           label={intl.formatMessage(messages.autoAnalysisSwitcherTitle)}
           description={Parser(intl.formatMessage(messages.analysisStatusInfo))}
-          format={Boolean}
+          format={this.formatAnalyzerAvailabilityValue}
```
Comment (DiscoElevator) [inline app/src/pages/inside/settingsPage/settingsTabs/autoAnalysisTab/analysisForm/strategyBlock/strategyBlock.jsx]: Could you please clarify what happens here?

File: app/src/controllers/project/utils.test.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { normalizeAttributesWithPrefix } from './utils';
+import { ANALYZER_ATTRIBUTE_PREFIX } from './constants';
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/controllers/project/utils.test.js]: Fixed

File: app/src/controllers/project/utils.test.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  };
+  const normalizedAttributes = {
+    [`${ANALYZER_ATTRIBUTE_PREFIX}.numberOfLogLines`]: '3',
+  };
+  describe('normalizeAttributesWithPrefix', () => {
+    test('should return {} in case of attributes object is empty', () => {
+      expect(normalizeAttributesWithPrefix({}, '')).toEqual({});
+      expect(normalizeAttributesWithPrefix({}, ANALYZER_ATTRIBUTE_PREFIX)).toEqual({});
+    });
+    test('should return normalized attributes object if attributes provided', () => {
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/controllers/project/utils.test.js]: Fixed

File: app/src/controllers/project/selectors.js [hunk 5]
```diff
@@ -17,6 +17,22 @@ export const userFiltersSelector = (state) => projectPreferencesSelector(state).
 
 export const defectTypesSelector = (state) => projectConfigSelector(state).subTypes || {};
 
+const attributesSelector = (state) => projectConfigSelector(state).attributes || {};
+
+export const attributesByPrefixSelector = (state, prefix) =>
+  createSelector(attributesSelector, (attributes) =>
+    Object.keys(attributes).reduce(
+      (result, attribute) =>
+        attribute.match(prefix)
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/controllers/project/selectors.js]: Done

File: app/src/pages/inside/settingsPage/settingsTabs/autoAnalysisTab/analysisForm/analysisForm.jsx (hunk 1 above)
Comment (AmsterGet) [reply app/src/pages/inside/settingsPage/settingsTabs/autoAnalysisTab/analysisForm/analysisForm.jsx]: Done

File: app/src/controllers/project/utils.test.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+import { ANALYZER_ATTRIBUTE_PREFIX } from './constants';
+
+describe('project/utils', () => {
+  const attributes = {
+    numberOfLogLines: '3',
+  };
+  const normalizedAttributes = {
+    [`${ANALYZER_ATTRIBUTE_PREFIX}.numberOfLogLines`]: '3',
+  };
+  describe('normalizeAttributesWithPrefix', () => {
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/controllers/project/utils.test.js]: Done

File: app/src/pages/inside/settingsPage/settingsTabs/autoAnalysisTab/analysisForm/strategyBlock/strategyBlock.jsx (hunk 2 above)
Comment (AmsterGet) [reply app/src/pages/inside/settingsPage/settingsTabs/autoAnalysisTab/analysisForm/strategyBlock/strategyBlock.jsx]: We get `boolean` attributes in the strings from server. So the `true`/`false` will be `"true"`/`"false"`

File: app/src/controllers/project/utils.js [hunk 7]
```diff
@@ -1,8 +1,10 @@
 export const normalizeAttributesWithPrefix = (attributes, prefix) =>
-  Object.keys(attributes).reduce(
-    (result, item) => ({
-      ...result,
-      [`${prefix}.${item}`]: attributes[item],
-    }),
-    {},
-  );
+  prefix
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/controllers/project/utils.js]: Done

File: app/src/controllers/project/utils.test.js [hunk 8]
```diff
@@ -1,22 +1,23 @@
...
       expect(normalizeAttributesWithPrefix({}, '')).toEqual({});
-      expect(normalizeAttributesWithPrefix({}, ANALYZER_ATTRIBUTE_PREFIX)).toEqual({});
+      expect(normalizeAttributesWithPrefix({}, prefix)).toEqual({});
     });
-    test('should return normalized attributes object if attributes provided', () => {
-      expect(normalizeAttributesWithPrefix(attributes, ANALYZER_ATTRIBUTE_PREFIX)).toEqual(
-        normalizedAttributes,
-      );
+    test('should return the same object if case of no prefix', () => {
+      expect(normalizeAttributesWithPrefix(attributes, '')).toEqual(attributes);
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/controllers/project/utils.test.js]: Fixed

**Pull request merged**